agent/php_environment.c

```diff
@@ -382,6 +382,134 @@ static void nr_php_gather_dispatcher_information(nrobj_t* env) {
 
   nro_set_hash_string(env, "Dispatcher", dstring);
 }
+void nr_php_process_environment_variable_to_nrobj(const char* prefix,
+                                                  const char* key,
+                                                  const char* value,
+                                                  nrobj_t* kv_hash) {
+  if ((NULL == prefix) || (NULL == kv_hash) || (NULL == key)) {
+    return;
+  }
+
+  if (nr_strlen(prefix) >= nr_strlen(key)) {
+    return;
+  }
+
+  if (0 == nr_strncmp(key, prefix, nr_strlen(prefix))) {
+    nro_set_hash_string(kv_hash, key, value);
+  }
+}
+
+char* nr_php_process_environment_variable_to_string(const char* prefix,
+                                                    const char* key,
+                                                    const char* value,
+                                                    char* kv_string,
+                                                    const char* kv_delimiter,
+                                                    const char* delimiter) {
+  if ((NULL == prefix) || (NULL == key)) {
+    return kv_string;
+  }
+
+  if (nr_strlen(prefix) >= nr_strlen(key)) {
+    return kv_string;
+  }
+
+  if (0 == nr_strncmp(key, prefix, nr_strlen(prefix))) {
+    kv_string = nr_str_append(kv_string, key + nr_strlen(prefix), delimiter);
```

Comment:
Looks like `nr_str_append()` will ignore `delimiter` if it is `NULL` but would it make sense to add a test to catch this case and handle it differently if needed?

Reply:
It doesn't ignore it, the design was that is null was passed in as a delimiter, nr_str_append, simply appends the strings together.  It does specifically check for NULL and if it detects it appends the strings together so that if a delimiter is passed in, it is used when appending the strings and if no delimiter is passed in, it just appends the strings.  Where were you thinking an additional test or different handling might be good?